packages/api/src/command/medical/patient/normalize-patient.ts

```diff
@@ -0,0 +1,60 @@
+import { PatientData, splitName } from "../../../domain/medical/patient";
+
+const commonDomainTypos: { [key: string]: string } = {
+  "gamil.com": "gmail.com",
+  // add more common typos here
+};
+
+export const normalizePatientData = (patient: PatientData): PatientData => {
+  const normalizedPatient: PatientData = {
+    ...patient,
+    // TODO: Handle the possibility of multiple patient names. right now we are just selecting for the first patient name.
+    firstName: splitName(patient.firstName.toLowerCase().replace(/['-]/g, ""))[0],
+    lastName: splitName(patient.lastName.toLowerCase().replace(/['-]/g, ""))[0],
+    contact: patient.contact?.map(contact => ({
+      ...contact,
+      email: contact.email ? normalizeEmail(contact.email) : contact.email,
+      phone: contact.phone ? normalizePhoneNumber(contact.phone) : contact.phone,
+    })),
+    address: patient.address.map(addr => ({
+      ...addr,
+      line1: addr.addressLine1
+        ? addr.addressLine1.toLowerCase().replace(/['-]/g, "")
+        : addr.addressLine1,
+      line2: addr.addressLine2
+        ? addr.addressLine2.toLowerCase().replace(/['-]/g, "")
+        : addr.addressLine2,
+      city: addr.city ? addr.city.toLowerCase().replace(/['-]/g, "").replace(/\s/g, "") : addr.city,
+      zip: addr.zip.slice(0, 5),
+    })),
+  };
+  return normalizedPatient;
+};
+
+function normalizeEmail(email: string): string {
+  let normalizedEmail = email.trim().toLowerCase();
+  // this needs to be better, but a lot of record linkage does email normalization
+  const domain = normalizedEmail.split("@")[1];
+  if (commonDomainTypos[domain]) {
+    normalizedEmail = normalizedEmail.replace(domain, commonDomainTypos[domain]);
+  }
+  return normalizedEmail;
+}
+
+// normalize phones
+function normalizePhoneNumber(phoneNumber: string): string {
```

Comment:
this isn't nessecary right now since the zod schema we have already enforces this. But we wont be able to use that zod schema to enforce what we get from CQ/CW. So maybe this can be taken out but the concept of needing to normalize what we get holds and is important

Reply:
Should also think about backwards compatibility. We have a lot of patients in our system currently without normalized data. Will need to do a migration?